File: Origin Answers/Python/Soru5.py
def strassen(dizi1: list[int],dizi2 : list[int]) -> list[list[int]]:
    p1 = dizi1[0][0] * (dizi2[0][1] - dizi2[1][1])
    p2 = (dizi1[0][0] + dizi1[0][1]) * dizi2[1][1]
    p3 = (dizi1[1][0] + dizi1[1][1]) * dizi2[0][0]
    p4 = dizi1[1][1] * (dizi2[1][0] - dizi2[0][0])
    p5 = (dizi1[0][0] + dizi1[1][1]) * (dizi2[0][0] + dizi2[1][1])
    p6 = (dizi1[0][1] - dizi1[1][1]) * (dizi2[1][0] + dizi2[1][1])
    p7 = (dizi1[0][0] - dizi1[1][0]) * (dizi2[0][0] + dizi2[0][1])
    sonucDizisi = [[0,0],[0,0]]
    sonucDizisi[0][0] = p5 + p4 - p2 + p6
    sonucDizisi[0][1] = p1 + p2
    sonucDizisi[1][0] = p3 + p4
    sonucDizisi[1][1] = p1 + p5 - p3 - p7
    return sonucDizisi



def main():
    inputArray = input("1. dizi icin 2x2 lik bir matris icin 4 sayi giriniz: ")

    dizi1 = inputArray.split()
    dizi1 = [int(i) for i in dizi1]
    dizi1 = [[dizi1[0],dizi1[1]],[dizi1[2],dizi1[3]]]
    inputArray = input("2. dizi icin 2x2 lik bir matris icin 4 sayi giriniz: ")

    dizi2 = inputArray.split()
    dizi2 = [int(i) for i in dizi2]
    dizi2 = [[dizi2[0],dizi2[1]],[dizi2[2],dizi2[3]]]
    sonucDizisi = strassen(dizi1,dizi2)

    for i in range(2):
        for j in range(2):
            print(sonucDizisi[i][j], end=" ")
        print()

main()

File: Competitor Answers/Güney KAYA/Soru_5.cs
using System;

namespace Hackaton_2
{
    class Program
    {
        static void Main(string[] args)
        {
            int[,] dizi1 = { { 3, 5 }, { 6, 8 } };
            int[,] dizi2 = { { 8, 3 }, { 4, 9 } };
            Strassen(dizi1, dizi2);
        }
        static void Strassen(int[,] a, int[,] b)
        {
            int p1 = a[0, 0] * (b[0, 1] - b[1, 1]);
            int p2 = (a[0, 0] + a[0, 1]) * b[1, 1];
            int p3 = (a[1, 0] + a[1, 1]) * b[0, 0];
            int p4 = a[1, 1] * (b[1, 0] - b[0, 0]);
            int p5 = (a[0, 0] + a[1, 1]) * (b[0, 0] + b[1, 1]);
            int p6 = (a[0, 1] - a[1, 1]) * (b[1, 0] + b[1, 1]);
            int p7 = (a[0, 0] - a[1, 0]) * (b[0, 0] + b[0, 1]);

            int[,] matris = new int[2, 2];

            matris[0, 0] = p5 + p4 - p2 + p6;
            matris[0, 1] = p1 + p2;
            matris[1, 0] = p3 + p4;
            matris[1, 1] = p1 + p5 - p3 - p7;

            for (int i = 0; i < 2; i++)
            {
                for (int j = 0; j < 2; j++)
                {
                    Console.Write(matris[i, j] + " ");
                }
                Console.Write("\n");
            }
        }
    }
}
